activiti-spring-process-extensions/src/test/java/org/activiti/spring/process/variable/DateFormatterProviderTest.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2019 Alfresco, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.activiti.spring.process.variable;
+
+import org.junit.Test;
+
+import java.time.DateTimeException;
+import java.time.Duration;
+import java.time.format.DateTimeParseException;
+import java.util.Date;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+
+public class DateFormatterProviderTest {
+
+    private DateFormatterProvider provider = new DateFormatterProvider("yyyy-MM-dd[['T'][ ]HH:mm:ss[.SSS'Z']]");
+
+    @Test
+    public void should_returnDate_when_stringRepresentsADate() {
+        //given
+        String dateStr = "1970-01-01";
+
+        //when
+        Date date = provider.toDate(dateStr);
+
+        //then
+        assertThat(date).hasTime(0);
+    }
+
+    @Test
+    public void should_returnDate_when_stringRepresentsADateWithTimeInformation() {
+        //given
+        String dateStr = "1970-01-01T01:01:01.001Z";
+        //calculate number of milliseconds after 1970-01-01T00:00:00.000Z
+        long time = Duration.ofHours(1).toMillis() + Duration.ofMinutes(1).toMillis() + Duration.ofSeconds(1).toMillis() + 1;
+
+        //when
+        Date date = provider.toDate(dateStr);
+
+        //then
+        assertThat(date).hasTime(time);
+    }
+
+    @Test
+    public void should_throwException_when_stringIsNotADate() {
+        //given
+        String dateStr = "this is not a date";
+
+        //when
+        assertThatExceptionOfType(DateTimeParseException.class)
+                .isThrownBy(() -> provider.parse(dateStr));
+    }
+
+    @Test
+    public void should_returnDate_when_longIsProvided() {
+        //given
+        long time = 1000;
+
+        //when
+        Date date = provider.toDate(time);
+
+        //then
+        assertThat(date).hasTime(time);
```

Comment:
should we make it clearer what we test in the "long" scenario? 
That we produce a valid date? That we produce a specific date value?